### DISCUSSION TITLE
GSOC project

### DISCUSSION THREAD

**Discussion by ayushtamra** [Hyperfoil/Horreum#354] [Q&A] [answered]
Hello maintainers, my name is Ayush and I am a student, I was looking to contribute as a participant in the GSOC 2023. In the idea list, it is mentioned that we have to migrate the project from react 17 to 18, is it the webapp directory we have to make changes on? Also it would be really helpful provide a slight elaboration about the idea and how can I start approaching it.
Thank you, Looking forward to contribute!

**Comment by jesperpedersen** [comment, marked as answer]
Thanks for your interest in GSoC 2023 !

Yes, this project is about upgrading the components used in the web stack to the latest technologies. You can get an overview from https://github.com/Hyperfoil/Horreum/blob/master/webapp/package.json on what is being used at the moment. The project should also document how to do future upgrades.

However, first you need to write your GSoC application for this project which can have multiple people submitting for the same slot.

Please, see the GSoC timeline here

 https://developers.google.com/open-source/gsoc/timeline

Thanks !

**Comment by ayushtamra** [comment]
Thanks @jesperpedersen , I will surely look into it
